fiber-test/fiber_test.ml

```diff
@@ -45,7 +45,7 @@ let test ?(expect_never = false) to_dyn f =
   let f =
     let on_error exn =
       Format.eprintf "%a@." Exn_with_backtrace.pp_uncaught exn;
-      Fiber.return ()
+      Exn_with_backtrace.reraise exn
```

Comment:
I don't see any this exception handled anywhere, am I missing something?

Reply:
it's just to make the test a little more resilient to fiber deadlocks. We can do one of two things:

* Wait for all fibers to complete and then print errors
* Print errors as they come

We're doing the latter, but without reraising the exceptions. That's a little fragile, but worked because we didn't rely on these exceptions in the tests.

Reply:
I'm confused: you seem to say that `Exn_with_backtrace.reraise` prints the error and does not reraise, but the name indicates that the exception *is* reraised

Reply:
The previous code printed the error but did not reraise. That was a mistake. The new code prints and reraises as expected.